app/controllers/katello/api/v2/package_groups_controller.rb

```diff
@@ -3,6 +3,52 @@ class Api::V2::PackageGroupsController < Api::V2::ApiController
     apipie_concern_subst(:a_resource => N_("a package group"), :resource => "package_groups")
     include Katello::Concerns::Api::V2::RepositoryContentController
 
+    before_action :find_repository, :only => [:create, :destroy]
+
+    api :POST, "/package_group", N_("Create a package group")
+    param :id, String, :required => true, :desc => N_("package group Pulp ID")
```

Comment:
Yeah, i think :id could be confusing, as the id that the object ends up having will be different.  I don't think i like the 'metadata' hash either, what about just:

```
      param :repository_id, String, :required => true, :desc => N_("repository ID")
      param :name, String, :desc => N_("package group name. Defaults to params[:id]")
      param :description, String, :required => true, :desc => N_("package group name. Defaults to params[:id]")
      param :user_visible, :bool, :desc => N_("set \"user_visible\" flag on package group. Defaults to true")
      param :mandatory_package_names, Array, :desc => N_("mandatory package names to include in the package group")
      param :optional_package_names, Array, :desc => N_("optional package names to include in the package group")
      param :conditional_package_names, Array, :desc => N_("conditional package names to include in the package group")
      param :default_package_names, Array, :desc => N_("default package names to include in the package group")
```